pysat/utils/io.py

```diff
@@ -172,121 +167,555 @@ def filter_netcdf4_metadata(inst, mdata_dict, coltype, remove=False,
+    trans_table : dict or NoneType
+        Keyed by current metalabels containing a list of
+        metadata labels to use within the returned dict. If None,
+        a default translation using `self.labels` will be used except
+        `self.labels.fill_val` will be mapped to `['_FillValue', 'FillVal']`
+        as required by netCDF files.
+
+    Returns
+    -------
+    export_dict : dict
+        A dictionary of the metadata for each variable of an output file
+
+    """
+
+    export_dict = {}
+
+    if trans_table is None:
+        trans_table = default_to_netcdf_translation_table(inst)
+
+    # Confirm there are no duplicated translation labels
+    trans_labels = []
+    for key in trans_table.keys():
+        trans_labels.extend(trans_table[key])
+
+    for i in np.arange(len(trans_labels)):
+        item = trans_labels.pop(0)
+        if item in trans_labels:
+            estr = ''.join(['There is a duplicated variable label value in ',
+                            '`trans_table`: ', item])
+            raise ValueError(estr)
+
+    # Translate each metadata label if a translation is provided
+    for key in meta_dict.keys():
+        export_dict[key] = {}
+        loop_meta_dict = meta_dict[key]
+        for orig_key in loop_meta_dict:
+            if orig_key in trans_table:
+                for translated_key in trans_table[orig_key]:
+                    export_dict[key][translated_key] = loop_meta_dict[orig_key]
             else:
-                meta_dict['Format'] = inst._get_var_type_code(coltype)
+                export_dict[key][orig_key] = loop_meta_dict[orig_key]
 
-            # Update the meta data
-            inst.meta[var] = meta_dict
-        else:
-            pysat.logger.info(''.join(('Unable to find MetaData for ', var)))
+    return export_dict
 
-    return
+
+def apply_table_translation_from_file(trans_table, meta_dict):
+    """Return `meta_dict` after applying `trans_table` to metadata keys.
+
+    Parameters
+    ----------
+    trans_table : dict
+       Mapping of metadata label used in a file to new value.
+    meta_dict : dict
+       Dictionary with metadata information from a loaded file.
+
+    Returns
+    -------
+    dict
+       `meta_dict` after the mapping in `trans_table` applied.
+
+    Note
+    ----
+    The purpose of this function is to maintain default compatibility
+    with `meta.labels` and existing code that writes and reads netcdf
+    files via pysat while also changing the labels for metadata within
+    the file.
+
+    """
+
+    filt_dict = {}
+    for var_key in meta_dict:
+        filt_dict[var_key] = {}
+
+        # Iterate over metadata for given variable `var_key`
+        for file_key in meta_dict[var_key].keys():
+            # Apply translation if defined
+            if file_key in trans_table:
+                new_key = trans_table[file_key]
+            else:
+                new_key = file_key
```

Comment:
Possible to do in one line, not sure if it would be too long, though.

Reply:
Just a bit too long.